### PR TITLE
Make Geyser track the client render distance and report it correctly to the Java server

### PR DESCRIPTION
**Pull request by rtm516**
Fixes: https://github.com/GeyserMC/Geyser/issues/741

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java [hunk 1]
```diff
@@ -120,7 +120,9 @@
 
     @Setter
     private Vector2i lastChunkPosition = null;
-    private int renderDistance;
+    private int serverRenderDistance;
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java:123]: Could you add some Javadoc comments here?

File: connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java [hunk 2]
```diff
@@ -487,16 +489,20 @@ public void sendForm(FormWindow window, int id) {
         windowCache.showWindow(window, id);
     }
 
-    public void setRenderDistance(int renderDistance) {
-        renderDistance = GenericMath.ceil(++renderDistance * MathUtils.SQRT_OF_TWO); //square to circle
-        if (renderDistance > 32) renderDistance = 32; // <3 u ViaVersion but I don't like crashing clients x)
-        this.renderDistance = renderDistance;
+    public void setServerRenderDistance(int serverRenderDistance) {
+        serverRenderDistance = GenericMath.ceil(++serverRenderDistance * MathUtils.SQRT_OF_TWO); //square to circle
+        if (serverRenderDistance > 32) serverRenderDistance = 32; // <3 u ViaVersion but I don't like crashing clients x)
```
Comment (DoctorMacc) [inline connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java:494]: We should probably get this squared away, or at least a better comment explaining when the crashing happens.

Comment (DoctorMacc) [reply connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java:498]: Tested on ViaVersion 3.0.2 dev on Java 1.8 with absolutely no problems with a render distance of 64.

**Comment by Hellohi3654** [comment]
I have noticed a few bugs with this PR
1. Chunks seem to load slower (Don't know why)
2. When a player is teleported the chunks don't load for them (you have to leave and rejoin for them to load)
3. Sometimes chunks completely refuse to load (This happens even though the server is at 20 TPS)

**Comment by DoctorMacc** [comment]
The third reason specifically (the second reason was yet to be documented) would be why this PR has been on hold.

**Comment by DoctorMacc** [comment]
I discussed this Shevchik (thank you if you lurk here!), and without further work this PR will not work the way we want it to. 

Bedrock unloads chunks whenever it feels like it. When its render distance is smaller than the server's, but the server sends all its chunks in its render distance, Bedrock unloads some/all of the outside chunks, and we have zero idea; the chunks are not updated again and they remain unloaded, causing holes. Therefore, we'd have to compute which chunks the client should see and send those when they are in range.

**Comment by soshimee** [comment]
Hmm, having this feature even for Java would be nice, for slow internets, we can just lower the render distance.
But I think Java doesn't send packets for client-side stuff unlike Bedrock.

**Comment by LegacyGamerHD** [comment]
is this still in hold is there any news about it

**Comment by Camotoy** [comment]
I'm going to close this as this needs a giant rework if we wanted to implement this in the server (see my November 25th 2020 comment). But, with plugins and Spigot forks implementing per-player view distance, I think our job on this one is going to be addressed.

**Pull request closed without merging**